concurrent_test.go

```diff
@@ -738,3 +754,135 @@ func validateSequential(rs historyRecords) error {
 
 	return nil
 }
+
+func TestConcurrentRepeatableRead(t *testing.T) {
+	t.Log("Preparing db.")
+	var (
+		bucket = []byte("data")
+		key    = []byte("mykey")
+
+		option = &bolt.Options{PageSize: 128}
+	)
+
+	db := mustCreateDB(t, option)
+	defer func() {
+		db.Close()
+	}()
+
+	// Create lots of K/V to allocate some pages
+	err := db.Update(func(tx *bolt.Tx) error {
+		b, err := tx.CreateBucketIfNotExists(bucket)
+		if err != nil {
+			return err
+		}
+		for i := 0; i < 1000; i++ {
+			k, v := fmt.Sprintf("key_%d", i), fmt.Sprintf("value_%d", i)
+			if err := b.Put([]byte(k), []byte(v)); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+	require.NoError(t, err)
+
+	// Remove all K/V to create some free pages
+	err = db.Update(func(tx *bolt.Tx) error {
+		b := tx.Bucket(bucket)
+		for i := 0; i < 1000; i++ {
+			k := fmt.Sprintf("key_%d", i)
+			if err := b.Delete([]byte(k)); err != nil {
+				return err
+			}
+		}
+		return b.Put(key, []byte("randomValue"))
+	})
+	require.NoError(t, err)
+
+	db = mustReOpenDB(t, db, option)
```

Comment:
Why re-opening the db is needed?

Reply:
To completely release all free pages.

Reply:
I see that `DB.freePages()` is only called inside `DB.beginRWTx()`. Can you give more details please, maybe add it as code comment?

Reply:
Added a comment.